### ISSUE TITLE
Better representation of new line and space characters in the translation-editor textarea

### ISSUE THREAD

**Issue by ib** [WeblateOrg/weblate#12568]
### Describe the problem

There are five characters whose representation is currently a bit confusing: newline, space, thin space, non-breaking space, and narrow non-breaking space.

First of all, there is an inconsistent display of line breaks and spaces between the source string and the translation-editor textarea (although the line break is not always displayed differently).

Furthermore, spaces are displayed the same as their non-breaking counterparts and are therefore indistinguishable.

### Describe the solution you would like

The display should be the same in both textareas and spaces should be distinguishable from their non-breaking counterparts.

Spaces and non-breaking spaces should be the same width, spaces with a dot in them, non-breaking spaces without.

Thin spaces and narrow non-breaking spaces should be smaller (as they already are, but perhaps a little wider, i.e. as wide as the current representation of the space), thin spaces with a dot in them, narrow non-breaking spaces without.

### Screenshots

**Inconsistency:**

![screenshot#1](https://github.com/user-attachments/assets/82bab2f9-101d-4f82-9c6d-fbbe84eceb7f)

**Indistinguishability:**
(showing: space,thin space,non-breaking space,narrow no-breaking space)

![screenshot#2](https://github.com/user-attachments/assets/4753cf87-89c2-484d-8d99-781bd3ac2021)

**Comment by nijel** [comment]
@meel-hd Any ideas how to make this betteR?

**Comment by meel-hd** [comment]
@nijel first they should be at least consistent.

**Comment by nijel** [comment]
Yes, but how to achieve that? The editor is a standard textarea with an overlay based on Prism.js doing syntax highlighting, so you can not really change the width of characters there as we do in Python based highlighting when showing the source.

**Comment by ib** [comment]
I don't want to sound like I know anything about implementation, but don't you already do something like this with the newline character? It usually (though not always) appears in the translation-editor textarea as:
![screenshot#3](https://github.com/user-attachments/assets/6f3c0db6-9378-414f-b847-e519d84b1f5a)

**Comment by meel-hd** [comment]
@nijel  maybe we can use a div with `contenteditable` attr set to true. It will make the div input behave like a textarea, but with the advantage that we can use `span` tags to style parts of the content as we want (highlight, replace, underline, give custom width, etc). At the end we can get the raw text entered by calling the method `innerText` or `textContent` on the div .

**Comment by ib** [comment]
If there is a way to achieve this, I suggest using `␣` (U+2423) to render non-breaking spaces.
